st2common/st2common/models/api/base.py

```diff
@@ -171,7 +172,17 @@ def callfunction(*args, **kwargs):
                 pecan.response.status = status_code
                 return json_encode(None)
 
-            result = f(*args, **kwargs)
+            try:
+                result = f(*args, **kwargs)
+            except TypeError as e:
+                message = str(e)
+                # Invalid number of arguments passed to the function meaning invalid path was
```

Comment:
Yes, the code is less ideal to put it nicely, but it works for now.

Eventually, when we fix kwargs abuse we should be able to solve in a slightly nicer manner by inspecting the method signature (we already kinda do this if `jsexpose` declares arg_types).